### ISSUE TITLE
Unable to link mupen64plus-video-n64.so using GCC 7.3

### ISSUE THREAD

**Issue by neilmunday** [ricrpi/mupen64plus-video-gles2n64#25]
Under both Fedora 26 (x86_64) and Arch Linux (Arm7) I cannot link `mupen64plus-video-n64.so` as it fails with the following linker errors (using GCC 7.3):

```
g++ -O3 -flto -fvisibility-inlines-hidden -DSDL_VIDEO_OPENGL_ES2=1 -DSDL_VIDEO_OPENGL=0     -O3 -flto -Wall -ffast-math -fno-strict-aliasing -fvisibility=hidden -I../../src -DSDL_VIDEO_OPENGL_ES2=1 -DSDL_VIDEO_OPENGL=0      -msse -fPIC -I/usr/include/libpng16  -I/usr/include/libdrm  -I/usr/include/SDL2 -D_REENTRANT -I../../../mupen64plus-core/src/api -O3  -MD -shared -Wl,-version-script,../../src/video_api_export.ver  _obj/ticks.o _obj/2xSAI.o _obj/3DMath.o _obj/Config.o _obj/CRC.o _obj/DepthBuffer.o _obj/F3DCBFD.o _obj/F3D.o _obj/F3DDKR.o _obj/F3DEX2.o _obj/F3DEX.o _obj/F3DPD.o _obj/F3DWRUS.o _obj/FrameSkipper.o _obj/GBI.o _obj/gDP.o _obj/gles2N64.o _obj/gSP.o _obj/L3D.o _obj/L3DEX2.o _obj/L3DEX.o _obj/N64.o _obj/OpenGL.o _obj/RDP.o _obj/RSP.o _obj/S2DEX2.o _obj/S2DEX.o _obj/ShaderCombiner.o _obj/Textures.o _obj/VI.o  -ldl -lpng16  -lGL  -lSDL2 -o mupen64plus-video-n64.so 
_obj/GBI.o (symbol from plugin): In function `GBI_Unknown(unsigned int, unsigned int)':
(.text+0x0): multiple definition of `One2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/GBI.o (symbol from plugin): In function `GBI_Unknown(unsigned int, unsigned int)':
(.text+0x0): multiple definition of `One2Four'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/GBI.o (symbol from plugin): In function `GBI_Unknown(unsigned int, unsigned int)':
(.text+0x0): multiple definition of `Two2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/GBI.o (symbol from plugin): In function `GBI_Unknown(unsigned int, unsigned int)':
(.text+0x0): multiple definition of `Three2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/GBI.o (symbol from plugin): In function `GBI_Unknown(unsigned int, unsigned int)':
(.text+0x0): multiple definition of `Three2Four'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/GBI.o (symbol from plugin): In function `GBI_Unknown(unsigned int, unsigned int)':
(.text+0x0): multiple definition of `Four2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/GBI.o (symbol from plugin): In function `GBI_Unknown(unsigned int, unsigned int)':
(.text+0x0): multiple definition of `Five2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/gDP.o (symbol from plugin): In function `gDP':
(.text+0x0): multiple definition of `One2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/gDP.o (symbol from plugin): In function `gDP':
(.text+0x0): multiple definition of `One2Four'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/gDP.o (symbol from plugin): In function `gDP':
(.text+0x0): multiple definition of `Two2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/gDP.o (symbol from plugin): In function `gDP':
(.text+0x0): multiple definition of `Three2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/gDP.o (symbol from plugin): In function `gDP':
(.text+0x0): multiple definition of `Three2Four'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/gDP.o (symbol from plugin): In function `gDP':
(.text+0x0): multiple definition of `Four2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/gDP.o (symbol from plugin): In function `gDP':
(.text+0x0): multiple definition of `Five2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/gles2N64.o (symbol from plugin): In function `PluginStartup':
(.text+0x0): multiple definition of `One2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/gles2N64.o (symbol from plugin): In function `PluginStartup':
(.text+0x0): multiple definition of `One2Four'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/gles2N64.o (symbol from plugin): In function `PluginStartup':
(.text+0x0): multiple definition of `Two2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/gles2N64.o (symbol from plugin): In function `PluginStartup':
(.text+0x0): multiple definition of `Three2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/gles2N64.o (symbol from plugin): In function `PluginStartup':
(.text+0x0): multiple definition of `Three2Four'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/gles2N64.o (symbol from plugin): In function `PluginStartup':
(.text+0x0): multiple definition of `Four2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/gles2N64.o (symbol from plugin): In function `PluginStartup':
(.text+0x0): multiple definition of `Five2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/gSP.o (symbol from plugin): In function `gSP':
(.text+0x0): multiple definition of `One2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/gSP.o (symbol from plugin): In function `gSP':
(.text+0x0): multiple definition of `One2Four'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/gSP.o (symbol from plugin): In function `gSP':
(.text+0x0): multiple definition of `Two2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/gSP.o (symbol from plugin): In function `gSP':
(.text+0x0): multiple definition of `Three2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/gSP.o (symbol from plugin): In function `gSP':
(.text+0x0): multiple definition of `Three2Four'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/gSP.o (symbol from plugin): In function `gSP':
(.text+0x0): multiple definition of `Four2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/gSP.o (symbol from plugin): In function `gSP':
(.text+0x0): multiple definition of `Five2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/OpenGL.o (symbol from plugin): In function `OGL_EnableRunfast()':
(.text+0x0): multiple definition of `One2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/OpenGL.o (symbol from plugin): In function `OGL_EnableRunfast()':
(.text+0x0): multiple definition of `One2Four'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/OpenGL.o (symbol from plugin): In function `OGL_EnableRunfast()':
(.text+0x0): multiple definition of `Two2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/OpenGL.o (symbol from plugin): In function `OGL_EnableRunfast()':
(.text+0x0): multiple definition of `Three2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/OpenGL.o (symbol from plugin): In function `OGL_EnableRunfast()':
(.text+0x0): multiple definition of `Three2Four'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/OpenGL.o (symbol from plugin): In function `OGL_EnableRunfast()':
(.text+0x0): multiple definition of `Four2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/OpenGL.o (symbol from plugin): In function `OGL_EnableRunfast()':
(.text+0x0): multiple definition of `Five2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/RSP.o (symbol from plugin): In function `RSP_LoadMatrix(float (*) [4], unsigned int)':
(.text+0x0): multiple definition of `One2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/RSP.o (symbol from plugin): In function `RSP_LoadMatrix(float (*) [4], unsigned int)':
(.text+0x0): multiple definition of `One2Four'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/RSP.o (symbol from plugin): In function `RSP_LoadMatrix(float (*) [4], unsigned int)':
(.text+0x0): multiple definition of `Two2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/RSP.o (symbol from plugin): In function `RSP_LoadMatrix(float (*) [4], unsigned int)':
(.text+0x0): multiple definition of `Three2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/RSP.o (symbol from plugin): In function `RSP_LoadMatrix(float (*) [4], unsigned int)':
(.text+0x0): multiple definition of `Three2Four'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/RSP.o (symbol from plugin): In function `RSP_LoadMatrix(float (*) [4], unsigned int)':
(.text+0x0): multiple definition of `Four2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/RSP.o (symbol from plugin): In function `RSP_LoadMatrix(float (*) [4], unsigned int)':
(.text+0x0): multiple definition of `Five2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/ShaderCombiner.o (symbol from plugin): In function `_color_param_str(int)':
(.text+0x0): multiple definition of `One2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/ShaderCombiner.o (symbol from plugin): In function `_color_param_str(int)':
(.text+0x0): multiple definition of `One2Four'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/ShaderCombiner.o (symbol from plugin): In function `_color_param_str(int)':
(.text+0x0): multiple definition of `Two2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/ShaderCombiner.o (symbol from plugin): In function `_color_param_str(int)':
(.text+0x0): multiple definition of `Three2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/ShaderCombiner.o (symbol from plugin): In function `_color_param_str(int)':
(.text+0x0): multiple definition of `Three2Four'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/ShaderCombiner.o (symbol from plugin): In function `_color_param_str(int)':
(.text+0x0): multiple definition of `Four2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/ShaderCombiner.o (symbol from plugin): In function `_color_param_str(int)':
(.text+0x0): multiple definition of `Five2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/Textures.o (symbol from plugin): In function `GetNone(void*, unsigned short, unsigned short, unsigned char)':
(.text+0x0): multiple definition of `Three2Four'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/Textures.o (symbol from plugin): In function `GetNone(void*, unsigned short, unsigned short, unsigned char)':
(.text+0x0): multiple definition of `One2Four'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/Textures.o (symbol from plugin): In function `GetNone(void*, unsigned short, unsigned short, unsigned char)':
(.text+0x0): multiple definition of `Three2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/Textures.o (symbol from plugin): In function `GetNone(void*, unsigned short, unsigned short, unsigned char)':
(.text+0x0): multiple definition of `One2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/Textures.o (symbol from plugin): In function `GetNone(void*, unsigned short, unsigned short, unsigned char)':
(.text+0x0): multiple definition of `Four2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/Textures.o (symbol from plugin): In function `GetNone(void*, unsigned short, unsigned short, unsigned char)':
(.text+0x0): multiple definition of `Five2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
_obj/Textures.o (symbol from plugin): In function `GetNone(void*, unsigned short, unsigned short, unsigned char)':
(.text+0x0): multiple definition of `Two2Eight'
_obj/Config.o (symbol from plugin):(.text+0x0): first defined here
collect2: error: ld returned 1 exit status
make: *** [Makefile:480: mupen64plus-video-n64.so] Error 1
```
I believe this is due to convert.h being included by multiple files and having function definitions etc. in the header file rather than a CPP file?

If I compile under CentOS 7.4 using GCC 4.8.5 it links fine.

**Comment by neilmunday** [comment]
Just to add, having compiled and installed GCC 5.4 for Arch Linux, mupen64plus-video-gles2n64 compiles and links fine also.

**Comment by redeamer** [comment]
I proposed a fix to the linking issue with #26. Would be glad if someone could review it.
